mender.go

```diff
@@ -104,20 +132,61 @@ func (m mender) GetCurrentImageID() string {
 	return imageID
 }
 
-func (m *mender) updateState() error {
+func (m *mender) changeState(state MenderState) {
+	log.Infof("Mender state: %v -> %v", m.state, state)
+	m.state = state
+}
+
+func (m *mender) hasUpgrade() (bool, error) {
 	env, err := m.ReadEnv("upgrade_available")
 	if err != nil {
-		return err
+		return false, err
 	}
 	upgradeAvailable := env["upgrade_available"]
 
 	// we are after update
 	if upgradeAvailable == "1" {
-		m.state = MenderRunningWithFreshUpdate
-		return nil
+		return true, nil
 	}
-	m.state = MenderFreshInstall
-	return nil
+	return false, nil
+}
+
+func (m *mender) updateState() {
+
+	newstate := MenderStateUnknown
+	var merr error
+
+	switch m.state {
```

Comment:
Please note that I might be wrong above, just want to make sure the logic is correct.